var/spack/repos/builtin/packages/opencv/package.py

```diff
@@ -394,6 +396,7 @@ class Opencv(CMakePackage, CudaPackage):
             default=False,
             description="Include opencv_{0} contrib module".format(mod),
         )
+        conflicts("+{0}".format(mod), when="~contrib")
```

Comment:
Since `contrib` is like a super variant, I think using a `when` makes more sense.
```
for mod in contrib_modules:
        variant(
            mod,
            when="+contrib",
            default=False,
            description="Include opencv_{0} contrib module".format(mod),
        )